File: Prog/Factoriser/Factovisor.c
#include <stdio.h>

unsigned long long factorial(int n) {
    if (n == 0)
        return 1;
    else
        return n * factorial(n - 1);
}

const char* is_divisible(int n, int m) {
    unsigned long long fact = factorial(n);
    if (fact % m == 0)
        return "divides";
    else
        return "does not divide";
}

int main() {
    int n, m;
    while (scanf("%d %d", &n, &m) != EOF) {
        printf("%d %s %d!\n", m, is_divisible(n, m), n);
    }
    return 0;
}

File: Prog/Factoriser/Factovisor.py
import math

def is_divisible(n, m):
    factorial = math.factorial(n)
    if factorial % m == 0:
        return f"{m} divides {n}!"
    else:
        return f"{m} does not divide {n}!"

# Read input until EOF (End of File)
while True:
    try:
        n, m = map(int, input().split())
        print(is_divisible(n, m))
    except EOFError:
        break
